src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Tasks/DoNotConfigureAwaitWithSuppressThrowing.cs

```diff
@@ -0,0 +1,72 @@
+            CreateLocalizableResourceString(nameof(DoNotUseConfigureAwaitWithSuppressThrowingMessage)),
+            DiagnosticCategory.Usage,
+            RuleLevel.BuildWarning,
+            CreateLocalizableResourceString(nameof(DoNotUseConfigureAwaitWithSuppressThrowingDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                // Only proceed if we have the Task<T>.ConfigureAwait(ConfigureAwaitOptions) method and if ConfigureAwaitOptions.SuppressThrowing is defined.
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask1, out var genericTask) ||
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksConfigureAwaitOptions, out var configureAwaitOptionsType) ||
+                    configureAwaitOptionsType.TypeKind != TypeKind.Enum ||
+                    genericTask
+                        .GetMembers("ConfigureAwait")
+                        .OfType<IMethodSymbol>()
+                        .Where(m => SymbolEqualityComparer.Default.Equals(m.ContainingType, genericTask) && m.Parameters.Length == 1 && SymbolEqualityComparer.Default.Equals(m.Parameters[0].Type, configureAwaitOptionsType))
+                        .FirstOrDefault() is not IMethodSymbol configureAwait ||
+                    configureAwaitOptionsType.GetMembers("SuppressThrowing").FirstOrDefault() is not IFieldSymbol suppressThrowing ||
+                    !DiagnosticHelpers.TryConvertToUInt64(suppressThrowing.ConstantValue, configureAwaitOptionsType.EnumUnderlyingType.SpecialType, out ulong suppressThrowingValue))
+                {
+                    return;
+                }
+
+                // Raise a diagnostic if the invocation is to Task<T>.ConfigureAwait with a constant value that includes SuppressThrowing
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    IInvocationOperation invocation = (IInvocationOperation)operationContext.Operation;
+
+                    if (SymbolEqualityComparer.Default.Equals(invocation.TargetMethod.OriginalDefinition, configureAwait) &&
+                        invocation.Arguments.Length == 1 &&
+                        invocation.Arguments[0].Descendants().FirstOrDefault(op => op.ConstantValue.HasValue && SymbolEqualityComparer.Default.Equals(op.Type, configureAwaitOptionsType)) is IOperation arg &&
```

Comment:
Instead of `invocation.Arguments[0].Descendants().FirstOrDefault(op => ...)`, could this be `invocation.Arguments[0].Value is {} op && ...`?